starlette/responses.py

```diff
@@ -73,7 +73,11 @@ def init_headers(self, headers: typing.Mapping[str, str] | None = None) -> None:
 
         content_type = self.media_type
         if content_type is not None and populate_content_type:
-            if content_type.startswith("text/"):
+            if (
+                content_type.startswith("text/")
+                and ";charset=" not in content_type.lower()
+                and "; charset=" not in content_type.lower()
```

Comment:
Can't it be any number of spaces between the `;` and `charset`? If so, only checking if charset is included is fine:
```suggestion
                and "charset" not in content_type.lower()
```

Reply:
i believe it actually could, good catch, I also didn't like the explicit verbosity of the space check. looking at the specs, `charset` should not be possible to be a substring any other place than as the optional parameter.

the following might be the safest:
```python
and "charset=" not in content_type.lower()
```